File: src/bumperbot_navigation/src/imu_republisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <sensor_msgs/msg/imu.hpp>

using std::placeholders::_1;
using namespace std::chrono_literals;

rclcpp::Publisher<sensor_msgs::msg::Imu>::SharedPtr imu_pub_;

void imuCallback(const sensor_msgs::msg::Imu &msg)
{
    sensor_msgs::msg::Imu imu;
    imu = msg;
    imu.header.frame_id = "base_footprint_ekf";
    imu_pub_->publish(imu);
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = rclcpp::Node::make_shared("imu_republisher");
    rclcpp::sleep_for(1s);
    imu_pub_ = node->create_publisher<sensor_msgs::msg::Imu>("/imu/ekf", 10);
    auto imu_sub_ = node->create_subscription<sensor_msgs::msg::Imu>("/imu/out", 10, imuCallback);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/bumperbot_localization/bumperbot_localization/imu_republisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Imu
import time;

class ImuRepublisher(Node):
    def __init__(self):
        super().__init__("imu_republisher")
        time.sleep(1)
        self.imu_sub_ = self.create_subscription(Imu, "/imu/out", self.imuCallback, 10)
        self.imu_pub_ = self.create_publisher(Imu, "/imu_ekf", 10)

    def imuCallback(self, msg):
        msg.header.frame_id = "base_footprint_ekf"
        self.imu_pub_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    imu_republisher = ImuRepublisher()
    rclpy.spin(imu_republisher)
    imu_republisher.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()